sqlserver/datadog_checks/sqlserver/schemas.py

```diff
@@ -0,0 +1,419 @@
+                        row["nullable"] = True
+                if table_id in id_to_table_data:
+                    id_to_table_data.get(table_id)["columns"] = id_to_table_data.get(table_id).get("columns", []) + [
+                        row
+                    ]
+                else:
+                    self._log.debug("Columns found for an unkown table with the object_id: {}".format(table_id))
+            else:
+                self._log.debug("Couldn't find id of a table: {}".format(table_name))
+        return len(data)
+
+    @tracked_method(agent_check_getter=agent_check_getter)
+    def _populate_with_partitions_data(self, table_ids, table_id_to_table_data, cursor):
+        rows = execute_query(PARTITIONS_QUERY.format(table_ids), cursor)
+        for row in rows:
+            table_id = row.pop("id", None)
+            if table_id is not None:
+                table_id_str = str(table_id)
+                if table_id_str in table_id_to_table_data:
+                    table_id_to_table_data[table_id_str]["partitions"] = row
+                else:
+                    self._log.debug("Partition found for an unkown table with the object_id: {}".format(table_id_str))
+            else:
+                self._log.debug("Return rows of [{}] query should have id column".format(PARTITIONS_QUERY))
+
+    @tracked_method(agent_check_getter=agent_check_getter)
+    def _populate_with_index_data(self, table_ids, table_id_to_table_data, cursor):
+        rows = execute_query(INDEX_QUERY.format(table_ids), cursor)
+        for row in rows:
+            table_id = row.pop("id", None)
+            if table_id is not None:
+                table_id_str = str(table_id)
+                if table_id_str in table_id_to_table_data:
+                    table_id_to_table_data[table_id_str].setdefault("indexes", [])
+                    table_id_to_table_data[table_id_str]["indexes"].append(row)
+                else:
+                    self._log.debug("Index found for an unkown table with the object_id: {}".format(table_id_str))
+            else:
+                self._log.debug("Return rows of [{}] query should have id column".format(INDEX_QUERY))
+
+    @tracked_method(agent_check_getter=agent_check_getter, track_result_length=True)
+    def _populate_with_foreign_keys_data(self, table_ids, table_id_to_table_data, cursor):
+        rows = execute_query(FOREIGN_KEY_QUERY.format(table_ids), cursor)
+        for row in rows:
+            table_id = row.pop("id", None)
+            if id is not None:
+                table_id_str = str(table_id)
+                if table_id_str in table_id_to_table_data:
+                    table_id_to_table_data.get(table_id_str).setdefault("foreign_keys", [])
+                    table_id_to_table_data.get(table_id_str)["foreign_keys"].append(row)
+                else:
+                    self._log.debug("Foreign key found for an unkown table with the object_id: {}".format(table_id_str))
```

Comment:
tag

Reply:
If we have this object id, what can we do with it? what is the next action a support engineer/dbm engineer would take if we see this in the logs?

Reply:
It will help understand that data for this table is not correct. 
The flow one can imagine - 
1. clients dojt see certain foreign keys
2. we ask them to run check manually, we see we had some internal errors with this table or not.

But again , I am now thinking - "let it throw" This should not happen unless we have a bug, and better to get notified right away.

The alternative - we continue collecting schemas while printing this log. But clients wont have insensitive to notify us and it can stay in code for longer and impact more people.

Reply:
To jojo's point, logging the object id is not quite useful because it's an internal id and not human readable. We could either log the foreign key name or just the `row` to give more info during debugging

Reply:
> To jojo's point, logging the object id is not quite useful because it's an internal id and not human readable. We could either log the foreign key name or just the row to give more info during debugging

+1

Reply:
logs removed , pls let me know if its not ok